pytensor/link/jax/dispatch/elemwise.py

```diff
@@ -74,12 +74,8 @@ def jax_funcify_DimShuffle(op, **kwargs):
     def dimshuffle(x):
         res = jnp.transpose(x, op.transposition)
 
-        shape = list(res.shape[: len(op.shuffle)])
-
-        for augm in op.augment:
-            shape.insert(augm, 1)
-
-        res = jnp.reshape(res, shape)
+        res = jax.lax.expand_dims(res, op.augment)
+        res = jax.lax.squeeze(res, op.drop)
```

Comment:
I think after the transpose the dropped dimension are all on the right, not in `op.drop`? So should be something like `squeeze(res, tuple(-(np.arange(len(op.drop) + 1))` ?

Also not related, but the `op.inplace` logic below is not needed in JAX, which never does inplace operations

Reply:
In the case of a transpose, `op.drop` would be an empty list. But `tuple(-(np.arange(len(op.drop) + 1))` would be non empty and if `squeeze` is done before `expand_dims`, the logic would fail in this case

Reply:
Sure, but an alternative is still needed. The current implementation is wrong AFAICT